packages/keystone/KSv5v6/v5/prepareKeystone.js

```diff
@@ -195,16 +198,33 @@ function prepareKeystone ({ onConnect, extendKeystoneConfig, extendExpressApp, s
 
         /** @type {(app: import('express').Application) => void} */
         configureExpress: (app) => {
+            const requestIdHeaderName = 'x-request-id'
+            const startRequestIdHeaderName = 'x-start-request-id'
 
             // NOTE(pahaz): we are always behind reverse proxy
             app.set('trust proxy', true)
 
+            app.use(function clearHeadersToPreventInjectionAttacks (req, res, next) {
+                try {
+                    for (const header in req.headers) {
+                        if (header === requestIdHeaderName || header === startRequestIdHeaderName) {
+                            validateRequestId(req.headers[header])
+                        } else {
+                            checkHeaderForInjection(req.headers[header], header)
+                        }
+                    }
+                    next()
```

Comment:
are you sure that you need to to call `nexi()` inside try / catch block?

Reply:
Done